server/lib/file-upload/aws.js

```diff
@@ -0,0 +1,76 @@
+import path from 'path'
+import aws from 'aws-sdk'
+
+import { FILE_MAX_AGE_MS } from './index.js'
+
+// Convert some of the more frequently used options to the AWS formatting
+function formatOptions(options = {}) {
+  const formatted = {}
+  if (options.cache) {
+    formatted.CacheControl = options.cache
+  }
+  if (options.type) {
+    formatted.ContentType = options.type
+  }
+  if (options.expires) {
+    formatted.Expires = options.expires
+  }
+
+  return formatted
+}
+
+// This is a generic implementation of a file-store using the aws-sdk. Note however, that it can be
+// used by any compatible storage provider, e.g. DigitalOcean Spaces or Amazon S3.
+export default class Aws {
+  constructor({ endpoint, accessKeyId, secretAccessKey, region, bucket }) {
+    const options = {
+      accessKeyId,
+      secretAccessKey,
+    }
+
+    // DO Spaces use endpoints
+    if (endpoint) {
+      options.endpoint = new aws.Endpoint(endpoint)
+    }
+    // Amazon S3 uses regions
+    if (region) {
+      options.region = region
+    }
+
+    this.bucket = bucket
+    this.client = new aws.S3(options)
+  }
+
+  _getNormalizedPath(filename) {
+    // Force posix path separators on aws-compatible services which use it to create faux folders.
+    const normalized = path.posix.normalize(filename)
+    if (path.isAbsolute(normalized) || normalized[0] === '.') {
+      throw new Error('Invalid directory')
+    }
+    return normalized
+  }
+
+  async write(filename, stream, options = {}) {
```

Comment:
Can you add a comment above this that specifies what valid options are/what their format is? (same for the other methods that take options). Doesn't need to be completely exhaustive, but at least what we convert from our style to AWS-API capitalization would be good